vlab-research/fly
# Feature/field interpolation

### What changes have you made?
- installed dependencies for Lodash and Mustache
- abstracted the interpolation logic from `replybot` to render dynamic values in the fields
- adapted the `getDynamicValue` function so that it works for the web survey
- set up the unit tests for `getDynamicValue`, `_interpolate` and `interpolateField`
- moved the biz logic into `ResponseStore` by creating a new method named `interpolationCheck` 
- changed the scope of `qa` at the component-level in order to make it accessible for interpolation – it now exists outside of the `handleSubmit` function

### Which issue(s) does this PR address?
#76 

### How to test
- `npm test` to check that all tests are passing
- `npm run dev` on `localhost:5000` to submit test answers to the survey 
- when landing on a question that requires a dynamic value the question should correctly interpolate by rendering with a previously-submitted answer
